src/dbnode/storage/fs_merge_with_mem.go

```diff
@@ -75,32 +76,33 @@ func (m *fsMergeWithMem) Read(
 	// it.
 	m.dirtySeriesToWrite[blockStart].Remove(element)
 
-	return m.fetchBlocks(ctx, element.Value, blockStart, nsCtx)
+	result, ok, err := m.fetchBlocks(ctx, element.Value, blockStart, nsCtx)
+	return result.Blocks, ok, err
```

Comment:
nit: is there any danger that `result` may become a nilable pointer in the future? Might be worth adding an explicit error check before accessing Blocks, but maybe thats a little paranoid

Reply:
+1. Yeah, probably doesn't hurt to do this.